### PR TITLE
SorrTask692_unit_test_get_session

### PR DESCRIPTION
**Pull request by Er1-x**
Task #692 
@samarth9008 @sonaalKant 
Create a session and create a new bucket with resource() and verify this new bucket with client().

Review request:
1. Am I test this boto3.session.Session() in a right way? Or maybe there is a formal way to verify that a Session is built up correctly.
2. Should I add some more test for other logic such as 'ck' ?
3. Is this ok to use resource() to test Session? I think this is a bit repetitive with #694

### CODE REVIEW

**Comment by Er1-x** [comment]
@samarth9008 I have run linter on this file, can you pls help me review and merge this PR?

File: helpers/test/test_haws.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        del os.environ["MOCK_AWS_DEFAULT_REGION"]
+
+    @mock_s3
+    def test_get_session1(self) -> None:
+        """
+        Test that `haws.get_session` correctly return a session without region
+        parameter.
+        """
+        aws_profile = "__mock__"
+
```
Comment (gpsaggese) [inline helpers/test/test_haws.py:41]: No need for empty lines.

File: helpers/test/test_haws.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    def set_up_test(self) -> None:
+        self.setUp()
+        os.environ["MOCK_AWS_ACCESS_KEY_ID"] = "mock_access_key"
+        os.environ["MOCK_AWS_SECRET_ACCESS_KEY"] = "mock_secret_access_key"
+        os.environ["MOCK_AWS_S3_BUCKET"] = "mock_s3_bucket"
+        os.environ["MOCK_AWS_DEFAULT_REGION"] = "us-east-1"
+
+    def tear_down_test(self) -> None:
+        del os.environ["MOCK_AWS_ACCESS_KEY_ID"]
```
[the remark this answers is not in the document]
Comment (Er1-x) [reply helpers/test/test_haws.py:29]: I think we need to tear down the mock system environment variables after the unit test. I'm not sure about this. I see similar code in test_s3.py

**Comment by Er1-x** [comment]
@gpsaggese sir, I have resolved all the comments, run linter and fix the `dassert_*` \ `self.assert*` problem you mentioned in #711 . pls check and verify.

**Comment by Er1-x** [comment]
> @gpsaggese sir, I have resolved all the comments, run linter and fix the `dassert_*` \ `self.assert*` problem you mentioned in #711 . pls check and verify.

Also, I have merged the master into my branch

File: helpers/test/test_haws.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (samarth9008) [reply helpers/test/test_haws.py:30]: Not needed you can remove it

**Review by samarth9008 — CHANGES_REQUESTED** [review]
TODO(Samarth): Don't merge before creating a PR on cmamp and testing it over there also

**Comment by Er1-x** [comment]
@samarth9008 @gpsaggese @sonaalKant
Fixed the nits and resolved the conflicts. Please verify

**Comment by Er1-x** [comment]
> TODO(Samarth): Don't merge before creating a PR on cmamp and testing it over there also

I see the same comments on the other PRs, should we create PRs on cmamp or should we wait for you to assign these tasks?

**Pull request merged**